integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/datamap/CarbonCreateDataMapCommand.scala

```diff
@@ -49,10 +61,21 @@ case class CarbonCreateDataMapCommand(
       throw new MalformedCarbonCommandException("Streaming table does not support creating datamap")
     }
     val LOGGER = LogServiceFactory.getLogService(this.getClass.getCanonicalName)
+    val dbName = tableIdentifier.database.getOrElse("default")
+    val tableName = tableIdentifier.table + "_" + dataMapName
 
-    if (dmClassName.equalsIgnoreCase(PREAGGREGATE.toString) ||
+    if (tableExists(sparkSession, dbName, tableName)) {
```

Comment:
tableExists method is called only from processMeta, so better not to create any method you can update the if statement
if(sparkSession.sessionState.catalog.listTables(dbName) .exists(_.table.equalsIgnoreCase(tableName)))

Reply:
ok,done